lib/KangarooTwelve.c

```diff
@@ -184,17 +187,17 @@ static unsigned int right_encode(unsigned char * encbuf, size_t value)
     return n + 1;
 }
 
-int KangarooTwelve_Initialize(KangarooTwelve_Instance *ktInstance, size_t outputByteLen)
+int KangarooTwelve_Initialize(KangarooTwelve_Instance *ktInstance, size_t outputByteLen, int securityLevel)
 {
     ktInstance->fixedOutputLength = outputByteLen;
     ktInstance->queueAbsorbedLen = 0;
     ktInstance->blockNumber = 0;
     ktInstance->phase = ABSORBING;
-    TurboSHAKE128_Initialize(&ktInstance->finalNode);
+    TurboSHAKE128_Initialize(&ktInstance->finalNode, securityLevel);
     return 0;
 }
 
-int KangarooTwelve_Update(KangarooTwelve_Instance *ktInstance, const unsigned char *input, size_t inputByteLen)
+int KangarooTwelve_Update(KangarooTwelve_Instance *ktInstance, const unsigned char *input, size_t inputByteLen, int securityLevel)
```

Comment:
I would put the security strength level in the ktInstance so that it does not have to be passed by the user every time.